### ISSUE TITLE
Auto relogin for edemo?

### ISSUE THREAD

**Issue by vans163** [ib-controller/ib-controller#70]
If you get disconnected from the IB server a popup box comes up asking you to confirm if you want to relogin.

**Comment by rlktradewright** [comment]
Really? I've been using IBController for nearly 13 years, during which time my connection to IB's servers has dropped out countless times, and I've never seen such a thing.

So you'll need to provide some evidence, or at least some information. Which version of TWS are you using? What are the circumstances? Can you make it happen at will? Can you supply a screenshot?

**Comment by vans163** [comment]
Interesting okay I screenshot/catch it next time.  I am using a edemo account with the IBGateway, the version recommended by ib-controller.

**Comment by rlktradewright** [comment]
The demo accounts don't behave in the same way as the live accounts in many respects, and this behaviour may be one.

IBController is not intended for use with the demo accounts. It does work with them on the whole, but no special code will be put into IBController to cope with things that happen in the demo system that don't happen in the live system.

**Comment by vans163** [comment]
Could be closed I guess.  Its just getting a live account you have to put a deposit, I think for developers they would first try IB and create working code, then put the money into the IB account and go live.

**Comment by rlktradewright** [comment]
Well yes, that's what I did back in 2003. I used the demo system for a couple of weeks, just enough to convince myself that the API met my needs. But it was fairly obviously not much use for anything else because there was no valid data, so I opened an account pretty quickly.

But there's no way I would recommend using the demo system to develop a trading system. Absolutely categorically not! If you're going to be risking real money, it's totally vital to be testing against at least the paper trading system (which is reasonably close to the live system in its behaviour). And you absolutely must test using real money with a live account before doing any kind of automated trading, or offering a product for third-party use. 

By the way, are you aware that when you use the demo system, you are allocated an account at random, but these accounts are shared. So while you're using the demo system, someone else logging on to it can be allocated the same account, and will receive a notification that the account is already in use from another computer, and giving them the option of forcing the other user (ie you!) off. Just this one feature, in my mind, makes the demo system completely useless for any serious development work.

The bottom line: forget the demo system...

**Comment by vans163** [comment]
Yea I noticed that the demo accs use different usernames internally like when you login using edemo to the server, it responds to the java gateway which demo account to use like edemo150, edemo55, etc..

Yea I realize developing a trading system you would most definitely need the paper account, also I think the demo account is throttled, I requested historical data and it seems off by price and volume from the actual data, also they don't give you over 7 days.  

My reasoning is that I can code to implement that API, for example using the Demo account I can make the code to place orders, check account balance, etc.  I guess I can just start the gateway manually to do that, since that is not a permanent thing.  

tl;dr: No need to support the demo account.
